src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs

```diff
@@ -173,18 +173,31 @@ public async Task<AuthFlowResponse> RespondToSmsMfaAuthAsync(RespondToSmsMfaRequ
         /// password required authentication challenge using an asynchronous call
         /// </summary>
         /// <param name="newPasswordRequest">RespondToNewPasswordRequiredRequest object containing the necessary 
+        /// <param name="requiredAttributes">Optional dictionnary of attributes that may be required by the user pool
+        /// Each attribute key must be prefixed by "userAttributes."
         /// parameters to respond to the current SMS MFA authentication challenge</param>
         /// <returns>Returns the AuthFlowResponse object that can be used to respond to the next challenge, 
         /// if one exists</returns>
-        public async Task<AuthFlowResponse> RespondToNewPasswordRequiredAsync(RespondToNewPasswordRequiredRequest newPasswordRequest)
```

Comment:
isn't this source compatible, but binary incompatible change?  Seems to me we can keep `RespondToNewPasswordRequiredAsync(RespondToNewPasswordRequiredRequest newPasswordRequest)` and introduce the new function?

Reply:
I have added a signature similar to the one already released. This should give 100% backward compatibility.